modules/db/src/main/java/org/mule/module/db/internal/domain/connection/DefaultDbConnection.java

```diff
@@ -393,22 +404,52 @@ public int getNetworkTimeout() throws SQLException
     @Override
     protected void resolveLobs(String typeName, Object[] attributes) throws SQLException
     {
-        try (ResultSet resultSet = this.getMetaData().getAttributes(this.getCatalog(), null, typeName, null))
+        Map<Integer, List> dataTypes = getDataTypes(typeName);
+
+        for(Integer key : dataTypes.keySet())
         {
+            List dataType = dataTypes.get(key);
+            doResolveLobIn(attributes, key, (int) dataType.get(0), (String) dataType.get(1));
+        }
+    }
 
+    private Map<Integer, List> getDataTypes(String typeName) throws SQLException
+    {
+        Map<Integer, List> dataTypes = new HashMap<>();
+
+        try (ResultSet resultSet = this.getMetaData().getAttributes(this.getCatalog(), null, typeName, null))
+        {
+            List<String> lobTypes = Arrays.asList(BLOB_DB_TYPE.getName(), CLOB_DB_TYPE.getName());
             int index = 0;
             while (resultSet.next())
             {
                 int dataType = resultSet.getInt(DATA_TYPE_INDEX);
                 String dataTypeName = resultSet.getString(ATTR_TYPE_NAME_INDEX);
 
-                doResolveLobIn(attributes, index, dataType, dataTypeName);
+                if(lobTypes.contains(dataTypeName))
+                {
+                    dataTypes.put(index, Arrays.asList(dataType, dataTypeName));
+                }
 
                 index++;
             }
         }
+        return dataTypes;
     }
 
+    protected void resolveLobsForArrays(String typeName, Object[] attributes) throws SQLException
+    {
+        Map<Integer, List> dataTypes = getDataTypes(typeName);
+
+        for(Integer key : dataTypes.keySet())
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Iterate over the "entrySet" instead of the "keySet". ([squid:S2864](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2864))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7681&issues=AWjt-QEVotIXnmsUG4vc&open=AWjt-QEVotIXnmsUG4vc)

[ISSUE_KEY]: AWjt-QEVotIXnmsUG4vc